solidity/contracts/converter/ConverterBase.sol

```diff
@@ -231,31 +213,54 @@ abstract contract ConverterBase is ConverterVersion, IConverter, TokenHolder, Co
     }
 
     /**
-     * @dev withdraws tokens held by the converter and sends them to an account
+     * @dev transfers reserves to a new converter during an upgrade
+     * can only be called by the converter upgraded which should be set at its owner
+     *
+     * @param _newConverter account to receive the new amount
+     */
+    function transferReservesOnUpgrade(address _newConverter)
+        external
+        override
+        protected
+        ownerOnly
+        only(CONVERTER_UPGRADER)
+    {
+        uint256 reserveCount = reserveTokens.length;
+        for (uint256 i = 0; i < reserveCount; ++i) {
+            IERC20 reserveToken = reserveTokens[i];
+
+            uint256 amount;
+            if (reserveToken == NATIVE_TOKEN_ADDRESS) {
+                amount = address(this).balance;
+            } else {
+                amount = reserveToken.balanceOf(address(this));
+            }
+
+            safeTransfer(reserveToken, _newConverter, amount);
+
+            syncReserveBalance(reserveToken);
+        }
+    }
+
+    /**
+     * @dev withdraws inactive reserves tokens held by the converter and sends them to an account
      * can only be called by the owner
-     * note that reserve tokens can only be withdrawn by the owner while the converter is inactive
-     * unless the owner is the converter upgrader contract
      *
-     * @param _token   ERC20 token contract address
-     * @param _to      account to receive the new amount
-     * @param _amount  amount to withdraw
+     * @param _token ERC20 token contract address
+     * @param _to account to receive the new amount
+     * @param _amount amount to withdraw
      */
-    function withdrawTokens(
+    function withdrawInactiveTokens(
```

Comment:
the name is a bit misleading, maybe just `withdrawReserve` and use the `inactive`, `validReserve` modifiers we already have instead of the inline logic?

alternatively - not support it at all :) any pool that's created through the registry (factory) is already active.

Reply:
I like the alternative better :)